lib/everypolitician/popolo/collection.rb

```diff
@@ -6,8 +6,21 @@ class Collection
       attr_reader :documents
       attr_reader :popolo
 
+      # set the class that represents individual items in the
+      # collection
+      def Collection.entity_class(entity)
+        @entity_class = entity
+      end
+
+      # we need to do this slightly awful looking thing becuase
+      # we want a class instance variable and not an instance
+      # variable
+      def entity_class
+        self.class.instance_variable_get :@entity_class
```

Comment:
I think you can probably get rid of this method entirely if you change the `self.entity_class` method above slightly so that it returns the entity class if one isn't given:

```ruby
def self.entity_class(entity = nil)
  @entity_class ||= entity
end
```

Then in `initialize` you can simply call `self.class.entity_class.new(p, popolo)`.